### PR TITLE
msg/async/rdma: Fix broken compilation

### PR DESCRIPTION
**Pull request by Adirl**
following api change in commit eb0f624

Signed-off-by: Sarit Zubakov <saritz@mellanox.com>

### CODE REVIEW

**Comment by Adirl** [comment]
@yuyuyu101

File: src/msg/async/rdma/RDMAServerSocketImpl.cc [hunk 1]
```diff
@@ -97,7 +97,7 @@ int RDMAServerSocketImpl::accept(ConnectedSocket *sock, const SocketOptions &opt
     ::close(sd);
     return -errno;
   }
-  net.set_priority(sd, opt.priority);
+  net.set_priority(sd, opt.priority, out->get_family());
```
Comment (yuyuyu101) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: I think we need to set priority after assigning value to "out"

Comment (saritz) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: Hi @yuyuyu101 ,
How about the following (change order put set_sockadde() before)?

`int RDMAServerSocketImpl::accept(ConnectedSocket *sock, const SocketOptions &opt, entity_addr_t *out, Worker *w)
{
  ldout(cct, 15) << __func__ << dendl;

  assert(sock);
  sockaddr_storage ss;
  socklen_t slen = sizeof(ss);
  int sd = ::accept(server_setup_socket, (sockaddr*)&ss, &slen);
  if (sd < 0) {
    return -errno;
  }
  ldout(cct, 20) << __func__ << " accepted a new QP, tcp_fd: " << sd << dendl;

  net.set_close_on_exec(sd);
  int r = net.set_nonblock(sd);
  if (r < 0) {
    ::close(sd);
    return -errno;
  }

  r = net.set_socket_options(sd, opt.nodelay, opt.rcbuf_size);
  if (r < 0) {
    ::close(sd);
    return -errno;
  }
-------------------------------------------------------------
  if (out)
     out->set_sockaddr((sockaddr*)&ss);
  net.set_priority(sd, opt.priority, out->get_family());
---------------------------------------------------------------

  RDMAConnectedSocketImpl* server;
  //Worker* w = dispatcher->get_stack()->get_worker();
  server = new RDMAConnectedSocketImpl(cct, infiniband, dispatcher, dynamic_cast<RDMAWorker*>(w));
  server->set_accept_fd(sd);
  ldout(cct, 20) << __func__ << " accepted a new QP, tcp_fd: " << sd << dendl;
  std::unique_ptr<RDMAConnectedSocketImpl> csi(server);
  *sock = ConnectedSocket(std::move(csi));
 
  return 0;
}
`

Comment (yuyuyu101) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: yes

Comment (saritz) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: Hi @yuyuyu101, @robbat2 , 
In the original commit: "msg/async: support IPv6 QoS. #13418" (https://github.com/ceph/ceph/pull/13418), the call to "set_priority()" is done without check if "out" is NULL. We need to first check if out is not NULL before sending it to "set_priority()". What should be default value if "out" is NULL?

For convenience I copied and marked the essential code to see the problem.

In PosixStack.cc::
260        int PosixServerSocketImpl::accept(ConnectedSocket *sock, const SocketOptions &opt, entity_addr_t *out, Worker *w) {
                …
281		handler.set_priority(sd, opt.priority, out->get_family()); 
…  
285		if (out)
286		     out->set_sockaddr((sockaddr*)&ss);
287		return 0;
288       } 

Thanks

@Adirl @orendu

Comment (yuyuyu101) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: fire a pr?

Comment (robbat2) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: @saritz i'm trying to see why `out` should EVER be null inside `PosixServerSocketImpl::accept`, I'd say we're probably better off with `assert(NULL != out)`

**Pull request merged**